ONSdigital/eq-author-app
# Disabled add routing button on last page of questionnaire

> BEFORE MAKING YOUR PR... There must be no linting errors and all tests must pass

### What is the context of this PR?

> Disabled the add routing button on for last page in questionnaire on front end
> When page becomes the last page in questionnaire any routing rules are deleted for that page, this can happen if a page or section is moved or deleted

### How to review 

> Add to the list below as appropriate, including screenshots when necessary

* Import the **Capability examples** questionnaire from pre-prod Author into your local environment and:
    * ensure it can be opened in Author;
    * then, ensure it can be viewed in Runner by pressing the **view survey** button
* Does this require a migration? We need one if existing JSON schema properties change
* Create questionnaire with two sections and two questions each
* Check you can't add routing to last question of questionnaire
* Add routing to first question in second section
* Move question to be last question and check routing has been deleted
* Add routing to first question in section two
* Delete last question and check new last question's routing has been deleted
* Add routing to last question of section one
* Move section two to become section one
* Check last question has no routing
* Move section back to be second section again and add routing to last question of section one
* Delete section two and check new last question has no routing

### What to do after everything is green

1. * [ ] Bring this branch up-to-date with Origin/Master
2. * [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. * [ ] Click the **Merge** button on this pull request
4. * [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. * [ ] Move the Jira ticket for this task into the next stage of the process
